src/components/dialogs/parameters/security-analysis-parameters.js

```diff
@@ -5,25 +5,248 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  */
 
-import React, { useCallback } from 'react';
-import { Grid } from '@mui/material';
+import React, { useCallback, useEffect, useState } from 'react';
+import { Grid, TextField, Tooltip } from '@mui/material';
+import InfoIcon from '@mui/icons-material/Info';
 import { CloseButton, DropDown, LabelledButton, useStyles } from './parameters';
 import { LineSeparator } from '../dialogUtils';
+import Typography from '@mui/material/Typography';
+import {
+    PARAM_SA_FLOW_PROPORTIONAL_THRESHOLD,
+    PARAM_SA_HIGH_VOLTAGE_ABSOLUTE_THRESHOLD,
+    PARAM_SA_HIGH_VOLTAGE_PROPORTIONAL_THRESHOLD,
+    PARAM_SA_LOW_VOLTAGE_ABSOLUTE_THRESHOLD,
+    PARAM_SA_LOW_VOLTAGE_PROPORTIONAL_THRESHOLD,
+} from '../../../utils/config-params';
+import { useIntl } from 'react-intl';
+
+const SecurityAnalysisFields = ({
+    label,
+    firstField,
+    secondField,
+    tooltipInfo,
+    initValue,
+    callback,
+    isSingleField,
+}) => {
+    const classes = useStyles();
+    const [values, setValues] = useState(initValue);
+
+    useEffect(() => {
+        setValues(initValue);
+    }, [initValue]);
+
+    const checkValue = useCallback((e, allowedRE, isPercentage) => {
+        const outputTransformToString = (value) => {
+            return value?.replace(',', '.') || '';
+        };
+        const newValue = e.target.value;
+        const isValid = allowedRE.exec(newValue);
+        const isAllValid = isPercentage ? isValid && newValue <= 100 : isValid;
+        if (isAllValid || newValue === '') {
+            setValues((prevState) => ({
+                ...prevState,
+                [e.target.name]: outputTransformToString(newValue),
+            }));
+        }
+    }, []);
+    const checkPerPercentageValue = useCallback(
+        (e) => {
+            const percentageRE = /^\d*[.,]?\d?\d?$/;
```

Comment:
percentageRE and doubleRE are the same. Could be the same const, and named "PositiveDoubleValue".